### PR TITLE
[chore](type cast) Fix some implicit cast

### PR DESCRIPTION
**Pull request by zhiqiang-hhhh**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Check List (For Committer)

- Test 

    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No colde files have been changed.
        - [ ] Other reason 

- Behavior changed:

    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?

    - [ ] No.
    - [ ] Yes. 

- Release note

    
    
    None

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

File: be/src/vec/data_types/convert_field_to_type.cpp [hunk 1]
```diff
@@ -257,7 +258,8 @@ void convert_field_to_typeImpl(const Field& src, const IDataType& type,
         JsonbWriter writer;
         Field::dispatch([&writer](const auto& value) { FieldVisitorToJsonb()(value, &writer); },
                         src);
-        *to = JsonbField(writer.getOutput()->getBuffer(), writer.getOutput()->getSize());
+        *to = JsonbField(writer.getOutput()->getBuffer(),
+                         static_cast<UInt32>(writer.getOutput()->getSize()));
```
Comment (HappenLee) [inline be/src/vec/data_types/convert_field_to_type.cpp:262]: why use `static_cast` not `cast_set`, `cast_set` can check overflow in DEBUG MODE

File: be/src/vec/data_types/data_type_time_v2.cpp [hunk 2]
```diff
@@ -37,6 +37,7 @@
 #include "vec/runtime/vdatetime_value.h"
 
 namespace doris {
+#include "common/compile_check_begin.h"
```
Comment (HappenLee) [inline be/src/vec/data_types/data_type_time_v2.cpp:40]: it's cpp why only begin not have end

File: be/src/vec/data_types/data_type.h [hunk 3]
```diff
@@ -58,8 +59,11 @@ class Field;
 using DataTypePtr = std::shared_ptr<const IDataType>;
 using DataTypes = std::vector<DataTypePtr>;
 constexpr auto SERIALIZED_MEM_SIZE_LIMIT = 256;
-inline size_t upper_int32(size_t size) {
-    return size_t((3 + size) / 4.0);
+
+template <typename T>
+inline T upper_int32(T size) {
```
Comment (HappenLee) [inline be/src/vec/data_types/data_type.h:64]: no need inline

File: be/src/vec/data_types/convert_field_to_type.cpp (hunk 1 above)
Comment (zhiqiang-hhhh) [reply be/src/vec/data_types/convert_field_to_type.cpp:262]: 1. Actually the impl of `cast_set` will do value check in all build type.
2. `convert_field_to_typeImpl` will be called on every tuple of the array. Don't want to introduce performance issue so I tried `cast_set<UInt32,UInt64, false>`
3. After I checked the impl of `cast_set`, I found `cast_set<UInt32,UInt64, false>` equals to static_cast so I use static_cast

File: be/src/vec/data_types/data_type_decimal.cpp [hunk 4]
```diff
@@ -124,7 +128,8 @@ void DataTypeDecimal<T>::to_string_batch_impl(const ColumnPtr& column_ptr,
             auto str = value.to_string(get_format_scale());
             chars.insert(str.begin(), str.end());
         }
-        offsets[row_num] = chars.size();
+
+        offsets[row_num] = static_cast<UInt32>(chars.size());
```
Comment (zhiqiang-hhhh) [inline be/src/vec/data_types/data_type_decimal.cpp:132]: cast by row, so not use cast_set for performance issue

**Comment by zhiqiang-hhhh** [comment]
run buildall

File: be/src/vec/data_types/data_type_time_v2.cpp (hunk 2 above)
Comment (zhiqiang-hhhh) [reply be/src/vec/data_types/data_type_time_v2.cpp:40]: according to https://selectdb.feishu.cn/wiki/FaJgwjAiuiVyKakCRQAcZmONnKe cpp file does not need the end header.

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run external

**Comment by zhiqiang-hhhh** [comment]
run external

File: be/src/vec/data_types/data_type_decimal.cpp [hunk 5]
```diff
@@ -230,7 +238,7 @@ const char* DataTypeDecimal<T>::deserialize(const char* buf, MutableColumnPtr* c
         buf = deserialize_const_flag_and_row_num(buf, column, &real_have_saved_num);
 
         // column data
-        auto mem_size = real_have_saved_num * sizeof(T);
+        UInt64 mem_size = real_have_saved_num * sizeof(T);
```
Comment (HappenLee) [inline be/src/vec/data_types/data_type_decimal.cpp:241]: if here do cast_set<> maybe we not need change line 251?

File: be/src/vec/data_types/data_type_number_base.h [hunk 6]
```diff
@@ -188,12 +189,12 @@ class DataTypeNumberBase : public IDataType {
         for (int row_num = 0; row_num < size; row_num++) {
             auto num = is_const ? col_vec.get_element(0) : col_vec.get_element(row_num);
             static_cast<const Derived*>(this)->push_number(chars, num);
-            offsets[row_num] = chars.size();
+            offsets[row_num] = cast_set<UInt32>(chars.size());
```
Comment (HappenLee) [inline be/src/vec/data_types/data_type_number_base.h:192]: seems we only need static_cast? push_number can check the chars is over uint32

File: be/src/vec/data_types/data_type_decimal.cpp (hunk 5 above)
Comment (zhiqiang-hhhh) [reply be/src/vec/data_types/data_type_decimal.cpp:241]: done

**Comment by zhiqiang-hhhh** [comment]
run buildall

File: be/src/vec/data_types/data_type_number_base.h (hunk 6 above)
Comment (zhiqiang-hhhh) [reply be/src/vec/data_types/data_type_number_base.h]: done

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by mrhhsg — APPROVED** [review]
LGTM

**Pull request merged**